## TDD evidence

RED commit: `ccee262ca`

Command:

```text
python -m pytest -q tests/test_story_regression.py::TestGracefulDegradation::test_empty_existing_tests_dir_does_not_start_nested_pytest
```

Observed failure:

```text
AssertionError: empty tests directory invoked pytest.main
Left contains one more item: --collect-only ... <empty tests directory>
1 failed
```

GREEN commit: `03f299bce`

The contract now covers both a literally empty directory and a directory containing only a non-test Python file; neither invokes nested pytest.

--- a/pdd/story_regression.py
+++ b/pdd/story_regression.py
@@ -51,6 +51,7 @@ _STORY_MARKER_HELP = (
 )
 
 PathLike = Union[str, Path]
+_STORY_TEST_GLOBS = ("test*.py", "*_test.py")
 
 
 def _default_tests_dir() -> Path:
@@ -293,7 +294,12 @@ def build_story_map(tests_dir: Optional[PathLike] = None) -> StoryTestMap:
     resolved_dir = Path(tests_dir) if tests_dir is not None else _default_tests_dir()
     if not resolved_dir.exists():
         return StoryTestMap({}, {})
-    if not any(path.is_file() for path in resolved_dir.rglob("test*.py")):
+    has_collection_candidate = any(
+        path.is_file()
+        for pattern in _STORY_TEST_GLOBS
+        for path in resolved_dir.rglob(pattern)
+    )
+    if not has_collection_candidate:
         return StoryTestMap({}, {})
 
     collector = _StoryCollector(resolved_dir)

--- a/tests/test_story_regression.py
+++ b/tests/test_story_regression.py
@@ -412,6 +412,28 @@ class TestGracefulDegradation:
             "tests directory without test*.py files invoked pytest.main"
         )
 
+    def test_pytest_suffix_named_candidate_still_starts_collection(
+        self, tmp_path: Path, monkeypatch
+    ):
+        tests_dir = tmp_path / "tests"
+        tests_dir.mkdir()
+        (tests_dir / "example_test.py").write_text(
+            "def test_example():\n    assert True\n", encoding="utf-8"
+        )
+        nested_pytest_calls = []
+
+        def _record_pytest_main(*args, **kwargs):
+            nested_pytest_calls.append((args, kwargs))
+            return 0
+
+        monkeypatch.setattr(story_regression.pytest, "main", _record_pytest_main)
+
+        build_story_map(tests_dir)
+
+        assert len(nested_pytest_calls) == 1, (
+            "pytest's default *_test.py discovery pattern must remain supported"
+        )
+
     def test_unparseable_module_is_skipped(self, tmp_path: Path):
         d = tmp_path / "tests"
         d.mkdir()

--- a/tests/test_story_regression.py
+++ b/tests/test_story_regression.py
@@ -434,6 +434,42 @@ class TestGracefulDegradation:
             "pytest's default *_test.py discovery pattern must remain supported"
         )
 
+    def test_direct_test_file_input_preserves_story_marker(self, tmp_path: Path):
+        test_file = tmp_path / "test_direct.py"
+        test_file.write_text(
+            "import pytest\n"
+            "@pytest.mark.story('direct_flow')\n"
+            "def test_direct():\n"
+            "    assert True\n",
+            encoding="utf-8",
+        )
+
+        smap = build_story_map(test_file)
+
+        assert smap.has_regression_test("direct_flow") is True
+
+    def test_custom_named_python_candidate_still_starts_collection(
+        self, tmp_path: Path, monkeypatch
+    ):
+        tests_dir = tmp_path / "tests"
+        tests_dir.mkdir()
+        (tests_dir / "custom_spec.py").write_text(
+            "def custom_case():\n    assert True\n", encoding="utf-8"
+        )
+        nested_pytest_calls = []
+
+        def _record_pytest_main(*args, **kwargs):
+            nested_pytest_calls.append((args, kwargs))
+            return 0
+
+        monkeypatch.setattr(story_regression.pytest, "main", _record_pytest_main)
+
+        build_story_map(tests_dir)
+
+        assert len(nested_pytest_calls) == 1, (
+            "custom pytest python_files candidates must not be pre-skipped"
+        )
+
     def test_unparseable_module_is_skipped(self, tmp_path: Path):
         d = tmp_path / "tests"
         d.mkdir()

